test/cases/DBVersioner.test.ts

```diff
@@ -0,0 +1,236 @@
+import assert from "node:assert";
+import {
+  DBVersioner,
+  MongoDBAdapter,
+  VERSION_TYPES,
+} from "../../src";
+import { loggerMock } from "../mocks/logger";
+import { dbMock, MongoClientMock } from "../mocks/mongo";
+
+
+describe("DB versioner", function () {
+  this.timeout(5000);
+
+  let mongoAdapter : MongoDBAdapter;
+  let versioner : DBVersioner;
+
+  const contractsVersion = "1.7.9";
+  const dbVersion = "109381236293746234";
+
+  before(async () => {
+
+    versioner = new DBVersioner({
+      dbVersion,
+      contractsVersion,
+      archive: false,
+      logger: loggerMock,
+    });
+
+    mongoAdapter = new MongoDBAdapter({
+      logger: loggerMock,
+      dbUri: "mongodb://mockedMongo",
+      dbName: "TestDatabase",
+      mongoClientClass: MongoClientMock,
+      versionerClass: DBVersioner,
+      dbVersion,
+      contractsVersion,
+    });
+
+    await mongoAdapter.initialize(dbVersion);
+  });
+
+  it("Should return existing temp version when calls configureVersioning()", async () => {
+    // db mock returns `DBVersion`
+    assert.equal(
+      await versioner.configureVersioning(dbMock),
+      dbVersion
+    );
+  });
+
+  it("Should make a DB version (Date.now()) when it's not exist", async () => {
+    // override the mock method to emulate the absence of a version in the DB
+    // @ts-ignore // because native .collection() requires arguments
+    dbMock.collection().findOne = async () => null;
```

Comment:
this is a weird way to do this. why? version is created upon instantiating a MongoAdapter. this is where you should control that. not in the DB itself

Reply:
you are checking a very different behaviour here. it's incorrect